pkgs/tools/security/tracee/default.nix

```
@@ -59,27 +56,33 @@ buildGoModule rec {
   # see passthru.tests.integration
   doCheck = false;
 
+  outputs = [ "out" "bin" "legacy" "lib" "share" ];
```

```suggestion
  outputs = [ "out" "lib" ];
```

That would probably easier to consume

---

I'm not sure how the split makes it harder to consume?

I kept the main "out" very useful having pretty much everything a user will need but giving separate outputs means if you know you only need the main `tracee` binary it's there under `tracee.bin`
Legacy output will disappear once that's removed in the future. Upstream maintainers have given the greenlight to remove it now though & shouldn't be a problem for our unstable channel so I can drop that early

I guess rather than linking I could depend on `outputsToInstall`?

---

The main benefit of multiple outputs is mostly to reduce closure size. If the separate outputs are links to the bigger one then they can't be installed separately. So it is more complicated without providing much gain. I think I'd prefer just `out` and `lib` too.

---

sure it can, you can just include `tracee.bin` in your system env or container or whatever

```
λ nix bundle --bundler github:NixOS/bundlers#toDockerImage github:06kellyjac/nixpkgs/tracee#tracee.bin

λ mv tracee-0.12.0.tar.gz tracee-bin-0.12.0.tar.gz

λ nix bundle --bundler github:NixOS/bundlers#toDockerImage github:06kellyjac/nixpkgs/tracee#tracee

λ du $(readlink ./tracee-0.12.0.tar.gz)
58616   /nix/store/1qsd41aq064wxfmlwpi4z5w6g9395b3r-tracee-0.12.0.tar.gz

λ du $(readlink ./tracee-bin-0.12.0.tar.gz)
32668   /nix/store/cc18zmmb9dkgg00mzyvlxlj1clrvagxb-tracee-0.12.0.tar.gz

λ docker load < tracee-0.12.0.tar.gz
2013c20b6873: Loading layer [==================================================>]  1.812MB/1.812MB
a9cb236e24b5: Loading layer [==================================================>]    297kB/297kB
6cf112bf428d: Loading layer [==================================================>]  30.77MB/30.77MB
90fce17ae0cf: Loading layer [==================================================>]  8.202MB/8.202MB
47e670efd9ae: Loading layer [==================================================>]  143.4kB/143.4kB
3ec45d13d845: Loading layer [==================================================>]  92.16kB/92.16kB
185b597ab364: Loading layer [==================================================>]  839.7kB/839.7kB
5cda2fe2b3ab: Loading layer [==================================================>]  1.126MB/1.126MB
fa4ea5799bfd: Loading layer [==================================================>]  3.666MB/3.666MB
51d514292f31: Loading layer [==================================================>]  1.772MB/1.772MB
58f6e5d6835f: Loading layer [==================================================>]  583.7kB/583.7kB
0276fd48123f: Loading layer [==================================================>]  122.9kB/122.9kB
d87b026ad8ad: Loading layer [==================================================>]  2.888MB/2.888MB
a119ce264191: Loading layer [==================================================>]  53.23MB/53.23MB
9fe0399b3a83: Loading layer [==================================================>]  77.61MB/77.61MB
2f15ef34242c: Loading layer [==================================================>]     10MB/10MB
a1d3f83dd160: Loading layer [==================================================>]  10.24kB/10.24kB
ddee65b264cb: Loading layer [==================================================>]  10.24kB/10.24kB
a4b9d21d882f: Loading layer [==================================================>]  10.24kB/10.24kB
Loaded image: tracee-0.12.0:latest

λ docker run -it tracee-0.12.0:latest tracee --version
Tracee version "v0.12.0"
λ docker run -it tracee-0.12.0:latest tracee-ebpf --version
Tracee version "v0.12.0"

λ docker load < ./tracee-bin-0.12.0.tar.gz
58f6e5d6835f: Loading layer [==================================================>]  583.7kB/583.7kB
51d514292f31: Loading layer [==================================================>]  1.772MB/1.772MB
0276fd48123f: Loading layer [==================================================>]  122.9kB/122.9kB
d87b026ad8ad: Loading layer [==================================================>]  2.888MB/2.888MB
a119ce264191: Loading layer [==================================================>]  53.23MB/53.23MB
faecf4444776: Loading layer [==================================================>]  10.24kB/10.24kB
The image tracee-0.12.0:latest already exists, renaming the old one with ID sha256:616a58ae3275261d869b6be79c8d6dc4cee891e2c9fe5ffeaf98bd8f102dde07 to empty string
Loaded image: tracee-0.12.0:latest

λ docker run -it tracee-0.12.0:latest tracee --version
Tracee version "v0.12.0"
λ docker run -it tracee-0.12.0:latest tracee-ebpf --version
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "tracee-ebpf": executable file not found in $PATH: unknown.
ERRO[0000] error waiting for container: context canceled
```

---

I have dropped the legacy bins because the tracee team advised it. Now there are less outputs but it's staying as-is. `tracee` can be used perfectly fine without the share dir, it's just a pack of rules that people may or may not use but it will come by default via `outputsToInstall` instead of linking now